### PR TITLE
Mod1 Week1 Independent Challenge_Ruthie Rabinovitch

### PR DESCRIPTION
**Pull request by rrabinovitch**

### CODE REVIEW

**Comment by timomitchel** [comment]
Great job making frequent commits throughout the challenge!

File: lib/school.rb [hunk 1]
```diff
@@ -1 +1,38 @@
+class School
+  attr_reader :start_time,
+              :hours_in_school_day,
+              :student_names
 
+  def initialize(start_time, hours_in_school_day)
+    @start_time = start_time
+    @hours_in_school_day = hours_in_school_day
+    @student_names = []
+  end
```
Comment (timomitchel) [inline lib/school.rb:10]: Nice job with this initialize method and the corresponding `attr_reader`s. This is exactly what we were hoping to see.

File: lib/school.rb [hunk 2]
```diff
@@ -1 +1,38 @@
...
+
+  def add_student_name(student_name)
```
Comment (timomitchel) [inline lib/school.rb:12]: This is a common pattern you will see throughout the mod. Nice job executing this.

File: lib/school.rb [hunk 3]
```diff
@@ -1 +1,38 @@
...
+    @student_names << student_name
+  end
+
+  def end_time
+    (@start_time.to_i + @hours_in_school_day).to_s + ':00'
```
Comment (timomitchel) [inline lib/school.rb:17]: Clean and effective.

File: lib/school.rb [hunk 4]
```diff
@@ -1 +1,38 @@
...
+  end
+
+  def is_full_time?
+    return true if @hours_in_school_day > 4
```
Comment (timomitchel) [inline lib/school.rb:21]: The guard statement is a bit redundant here. We only need `@hours_in_school_day > 4` in this method as that statement will always return a boolean literal.

File: lib/school.rb [hunk 5]
```diff
@@ -1 +1,38 @@
...
+    false
+  end
+
+  def standard_student_names
+    @student_names.map do |name|
```
Comment (timomitchel) [inline lib/school.rb:26]: The perfect use case for a `#map`

File: lib/school.rb [hunk 6]
```diff
@@ -1 +1,38 @@
...
+      name.capitalize
+    end
+  end
+
+  def convert_end_time_to_clock_time
```
Comment (timomitchel) [inline lib/school.rb:31]: Great job setting up this conditional to get the job done here!

File: test/school_test.rb [hunk 7]
```diff
@@ -45,11 +40,38 @@ def test_it_can_add_student_names
...
-    skip
     school1 = School.new('9:00', 7)
     school2 = School.new('9:00', 3)
 
     assert_equal '16:00', school1.end_time
     assert_equal '12:00', school2.end_time
   end
+
+  #Iteration 3 Tests:
+  def test_it_can_identify_whether_full_time
```
Comment (timomitchel) [inline test/school_test.rb:51]: Nice job testing both a true and false return for this boolean method!

File: test/school_test.rb [hunk 8]
```diff
@@ -45,11 +40,38 @@ def test_it_can_add_student_names
...
+    school1 = School.new('9:00', 7)
+    school2 = School.new('9:00', 3)
+
+    assert_equal true, school1.is_full_time?
+    assert_equal false, school2.is_full_time?
+  end
+
+  def test_it_can_standardize_student_names
+    school = School.new('9:00', 7)
```
Comment (timomitchel) [inline test/school_test.rb:60]: A beautiful test setup and execution here!

File: test/school_test.rb [hunk 9]
```diff
@@ -45,11 +40,38 @@ def test_it_can_add_student_names
...
+
+    school.add_student_name('Aurora')
+    school.add_student_name('tim')
+    school.add_student_name('megan')
+
+    assert_equal ['Aurora', 'Tim', 'Megan'], school.standard_student_names
+  end
+
+  #Iteration 4 Tests:
+  def test_it_can_convert_end_time_to_clock_time #convert from 24hr to 12hr format
```
Comment (timomitchel) [inline test/school_test.rb:70]: Great job robustly testing both the `if` and `else` blocks of your conditional!

**Comment by rrabinovitch** [comment]
Thanks for the thorough feedback @timomitchel 
The challenge review the other day helped me realize the redundancy of that guard statement in the `is_full_time?` method - I'll keep that in mind next time around!

**Pull request closed without merging**